### PR TITLE
Fixed the EnvUtilTest to be compatible with Windows OS and OpenJDK

### PR DESCRIPTION
**Pull request by saravanan-kandaswamy**
Please answer these questions before submitting your issue.

- Why do you submit this issue?
-  Bug
The Unit test EnvUtilTest.java case in apm-datacarrier module is not passing in Windows OS 10. 

___
### Question
- What do you want to know?
Can we build the code in Windows Machine?
___
### Bug
- Which version of SkyWalking, OS and JRE?
Skywalking - Latest Code base as of Master
OS - Windows 10
JRE - tried with Open JDK from Redhat and from Oracle JDK 8
Open JDK:
openjdk version "1.8.0_201-2-redhat"
OpenJDK Runtime Environment (build 1.8.0_201-2-redhat-b09)

Oracle JDK
java version "1.8.0_211"
Java(TM) SE Runtime Environment (build 1.8.0_211-b12)

- Which company or project?
I am trying the build source code in local system to write more unit tests, however the build is failing before that. 
- What happen?
when I try to build apm-datacarrier the build fails with the following error

[INFO] Running org.apache.skywalking.apm.commons.datacarrier.EnvUtilTest
[ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.013 s <<< FAILURE! - in org.apache.skywalking.apm.commons.datacarrier.EnvUtilTest
[ERROR] getInt(org.apache.skywalking.apm.commons.datacarrier.EnvUtilTest)  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.AssertionError: expected:<123> but was:<234>
        at org.apache.skywalking.apm.commons.datacarrier.EnvUtilTest.getInt(EnvUtilTest.java:56)

[ERROR] getLong(org.apache.skywalking.apm.commons.datacarrier.EnvUtilTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: expected:<12345678901234567> but was:<123>
        at org.apache.skywalking.apm.commons.datacarrier.EnvUtilTest.getLong(EnvUtilTest.java:62)

[INFO] Running org.apache.skywalking.apm.commons.datacarrier.partition.ProducerThreadPartitionerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.013 s - in org.apache.skywalking.apm.commons.datacarrier.partition.ProducerThreadPartitionerTest
[INFO] Running org.apache.skywalking.apm.commons.datacarrier.partition.SimpleRollingPartitionerTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.013 s - in org.apache.skywalking.apm.commons.datacarrier.partition.SimpleRollingPartitionerTest
[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   EnvUtilTest.getInt:56 expected:<123> but was:<234>
[ERROR]   EnvUtilTest.getLong:62 expected:<12345678901234567> but was:<123>
[INFO]
[ERROR] Tests run: 17, Failures: 2, Errors: 0, Skipped: 0 

___
### Requirement or improvement
After little bit of search I found that the code doesn't work on Windows to make it work we need to follow the steps as mentioned in link:
https://stackoverflow.com/a/38073822

I have created the pull request with the necessary change.
Kindly review the same and let me know the results. 

Thanks and Regards,
Saravanan Kandaswamy

### CODE REVIEW

File: apm-commons/apm-datacarrier/pom.xml [hunk 1]
```diff
@@ -26,5 +26,17 @@
...
 
     <artifactId>apm-datacarrier</artifactId>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
+                    <target>8</target>
```
Comment (wu-sheng) [inline apm-commons/apm-datacarrier/pom.xml]: The common lib can't compile in 1.8, because the agent should be jdk 1.6 compatible.

Comment (saravanan-kandaswamy) [reply apm-commons/apm-datacarrier/pom.xml]: I have used Lambda in unit test, will try to change with unit test and make it compatible with JDK 1.6

**Comment by wu-sheng** [comment]
Could you try to add more jdk and OS versions to travis yml?

**Comment by wu-sheng** [comment]
Maybe I need to set other CI platform to support windows compile. @liuhaoyang

**Comment by wu-sheng** [comment]
@saravanan-kandaswamy Please help us verify this PR. I don't have windows in many jdk versions at local. Need you work on CI to support those and avoid this in the future.

**Comment by saravanan-kandaswamy** [comment]
> 
> 
> @saravanan-kandaswamy Please help us verify this PR. I don't have windows and many jdk versions at local. Need you work on CI to support those and avoid this in the future.

I have no knowledge on Travis-CI, i do know Jenkins though, let me know what kind of activity i need to do to setup the CI for windows.

**Comment by liuhaoyang** [comment]
> Maybe I need to set other CI platform to support windows compile. @liuhaoyang

we use the appveyor platform to support windows compile. We have previously created an appveyor account for skyapm-dotnet and I will use it for testing.

**Comment by wu-sheng** [comment]
@liuhaoyang https://issues.apache.org/jira/browse/INFRA-18330 JIRA ticket submitted. Asking Apache INFRA to open it.

**Comment by wu-sheng** [comment]
OpenJDK passed in CI https://github.com/apache/skywalking/pull/2583

**Comment by wu-sheng** [comment]
Fixed in #2593

**Pull request closed without merging**